src/context/SessionInfoContext.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { createContext, useCallback, useContext, useReducer, useState, useEffect } from 'react'
+
+import { useCurrentWallet, useSetCurrentWallet, useSetCurrentWalletInfo } from './WalletContext'
+import { useWebsocket, CJ_STATE_TAKER_RUNNING, CJ_STATE_MAKER_RUNNING } from './WebsocketContext'
+import { clearSession } from '../session'
+
+import * as Api from '../libs/JmWalletApi'
+
+// interval in milliseconds for periodic session requests
+const SESSION_REQUEST_INTERVAL = 10_000
+
+const SessionInfoContext = createContext()
+
+const SessionInfoProvider = ({ children }) => {
+  const currentWallet = useCurrentWallet()
```

Comment:
> Do we still need this to be a context? Components will use session context instead of the wallet context now right? If that's the case, we could move the current wallet states into the session context.

Hmm.. I think I know what you are referring to, but I am not sure. `SessionInfoContext` only stores and updates information from the `/session` endpoint or from websocket data. The `WalletContext` "stores" the auth token and the corresponding wallet name is was issued for. So they are not _really_ the same and I am unsure whether they should be combined. The `SessionInfoContext` can be used by components that do not need a wallet to be active. 

Maybe the wording is not optimal and `SessionInfo` should be renamed as it has nothing to do with a local session from a user's perspective - `ServiceInfo`, `ServerState`, etc... Would something like that make it more clear?

> Same for the websockets probably.

Yes, if `WalletContext` and `SessionInfoContext` are combined, the `WebsocketContext` must also be taken care of, as `SessionInfoContext` depends on `WebsocketContext` which needs the `WalletContext` to enable receiving data in the first place.

I tried to keep the context as "small as possible".
Should everything be handled in a single context?

Reply:
> Maybe the wording is not optimal and SessionInfo should be renamed as it has nothing to do with a local session from a user's perspective - ServiceInfo, ServerState, etc... Would something like that make it more clear?

👌  Let's do it like that and keep the wallet and socket components external and modular.